mailbox/store/src/main/java/org/apache/james/mailbox/store/StoreMailboxManager.java

```diff
@@ -700,9 +700,12 @@ private boolean hasChildIn(Mailbox parentMailbox, List<Mailbox> mailboxesWithPat
     }
 
     private Stream<MailboxId> getInMailboxes(ImmutableSet<MailboxId> inMailboxes, MailboxSession session) throws MailboxException {
-       if (inMailboxes.isEmpty()) {
+        if (inMailboxes.isEmpty()) {
```

Comment:
I'd like to discuss the relevance of doing this `[]` sanitizing at the mailbox level.

This looks like a JMAP specific concern thus I would be in favor of handling this at the JMAP level.

(Other pull request?)

Reply:
In the meantime, perhaps it makes no sense to have 'empty' mailboxes when we do a search with a system session here.

Can I suggest throwing an illegalArgumentException?